SREP-1539: Ready for app-sre pipeline

lisa opened the pull request:
Deletes deploy/* to follow other pattern, but leaves behind a file to
keep the directory around.

Signed-off-by: Lisa Seelye <lseelye@redhat.com>

lisa committed:
SREP-1539: Ready for app-sre pipeline

Deletes deploy/* to follow other pattern, but leaves behind a file to
keep the directory around.

Signed-off-by: Lisa Seelye <lseelye@redhat.com>

lisa committed:
SREP-1539: Add pipeline scripts

Signed-off-by: Lisa Seelye <lseelye@redhat.com>

lisa committed:
SREP-1539: remove redundant mkdir

Signed-off-by: Lisa Seelye <lseelye@redhat.com>

rogbas commented on hack/00-osd-managed-prometheus-exporter-machine-api.selectorsyncset.yaml.tmpl:
```diff
@@ -0,0 +1,234 @@
+apiVersion: v1
```
Isn't that file generated dynamically? Does it make sense to keep it in the repo?

lisa replied:
The other exporters seem to have it commited?

jewzaam replied:
This file is what saasherder pulls and uses to configure things on hive.  It has to exist and can't be built dynamically since the jenkins job pulls the raw file using curl or wget or something.

jewzaam commented on scripts/templates/template.yaml:
```diff
@@ -0,0 +1,22 @@
+metadata:
+  name: selectorsyncset-template
+objects:
+# replace old SSS w/ nothing so we don't have manual cleanup to do
```
Everything past line #12 in this file needs to be deleted.  This repo doesn't have some legacy named SSS deployed that needs clobbered.

jewzaam commented on scripts/templates/template.yaml:
```diff
@@ -0,0 +1,22 @@
+  required: true
+metadata:
+  name: selectorsyncset-template
+objects:
```
```suggestion
objects: []
```

jewzaam commented on .gitignore:
```diff
@@ -1,3 +1,6 @@
+# Rendered templates
+deploy/
```
You ignore it and added .gitkeep?  Keeping the contents in deploy would let us point to that for an oc apply if someone else wanted to use it.  Else it's use oc process on the template or build it yourself.  All doable.  Not a blocker for this PR, just something I wondered about.

lisa replied:
Yeah... we need the directory because that's where non-Hive artifacts land, but we do not want them in git. This is a pattern used in other places (which doesn't mean it's good or bad, but it's "the pattern").

lisa committed:
SREP-1539: Address PR feedback

Lines 12+ were cargo culted needlessly, and `objects` should be an empty
list `[]`

Signed-off-by: Lisa Seelye <lseelye@redhat.com>

lisa committed:
SREP-1539: re-run make

Signed-off-by: Lisa Seelye <lseelye@redhat.com>
